api/payment/PaymentApi.php

```diff
@@ -55,29 +56,37 @@ class PaymentApi
      */
     protected $verificationToken = '';
 
+    /**
+     * Payment billing info.
+     *
+     * @var array
+     */
+    protected $billing_info = [];
+
 
     /**
      * CancelOrder constructor.
      *
      * @param EEG_SquareOnsite $gateway
      * @param SquareApi        $api
+     * @param array            $billing_info
      * @param string           $payment_token
      * @param string|null      $verificationToken
      * @param int              $TXN_ID
      */
     public function __construct(
         EEG_SquareOnsite $gateway,
         SquareApi $api,
-        string $payment_token,
-        ?string $verificationToken,
+        array $billing_info,
         int $TXN_ID
     ) {
         $this->api               = $api;
         $this->gateway           = $gateway;
-        $this->payment_token     = $payment_token;
-        $this->verificationToken = $verificationToken ?? '';
+        $this->payment_token     = $billing_info['eea_square_token'];
```

Comment:
golden rule of programming: if it can be null, it will be null.

```suggestion
        $this->payment_token     = $billing_info['eea_square_token'] ?? '';
```